File: 02_paradigm_exploration/math_and_miscellaneous/prime.scala
object PrimeChecker {
  def isPrime(n: Int): Boolean = {
    if (n <= 1) false
    else !(2 to math.sqrt(n).toInt).exists(n % _ == 0)
  }

  def main(args: Array[String]): Unit = {
    println(isPrime(17))  // Output: true
  }
}

File: 02_paradigm_exploration/math_and_miscellaneous/prime.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	sqrt := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrt; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println(isPrime(17))  // Output: true
}
